model/user.go

```diff
@@ -51,6 +51,23 @@ func PQArray2Roles(roleArray *pq.StringArray) Roles {
 	return res
 }
 
+func Array2Roles(roles []string) Roles {
+	res := Roles{}
+	for _, roleString := range roles {
+		res = append(res, String2Role(roleString))
+	}
+	return res
+}
+
+func RoleInRoles(role Role, roles Roles) bool {
```

Comment:
It's not so elegant to write the function in this way. Try implement 'include' for 'Role' next time.